tracing-subscriber/src/reload.rs

```diff
@@ -144,6 +145,19 @@ where
     fn max_level_hint(&self) -> Option<LevelFilter> {
         try_lock!(self.inner.read(), else return None).max_level_hint()
     }
+
+    #[doc(hidden)]
+    unsafe fn downcast_raw(&self, id: TypeId) -> Option<NonNull<()>> {
+        // Safety: It is generally unsafe to downcast through a reload, because
+        // the pointer can be invalidated after the lock is dropped.
+        // `NoneLayerMarker` is a special case because it
+        // is never dereferenced.
```

Comment:
it might also be worth mentioning that even if someone *does* somehow decide to dereference the `NoneLayerMarker`, it _is_ a valid pointer to the `NONE_LAYER_MARKER` static, and the pointer itself won't be invalidated even if the subscriber is reloaded (since the pointer is not actually pointing to the subscriber that's inside the lock)...

Reply:
maybe:
```suggestion
        // Safety: it is generally unsafe to downcast through a reload, because
        // the pointer can be invalidated after the lock is dropped.
        // `NoneLayerMarker` is a special case because it
        // is never dereferenced.
        //
        // Additionally, even if the marker type *is* dereferenced (which it
        // never will be), the pointer should be valid even if the subscriber
        // is reloaded, because all `NoneLayerMarker` pointers that we return
        // actually point to the global static singleton `NoneLayerMarker`,
        // rather than to a field inside the lock.
```